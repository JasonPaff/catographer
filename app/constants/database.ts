export const databaseName = "catographer-db";
