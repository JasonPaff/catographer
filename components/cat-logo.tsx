import { Image, type ImageProps } from "tamagui";

export function CatLogo(props: Omit<ImageProps, "source" | "objectFit">) {
	return (
		<Image
			height={"$2.5"}
			objectFit={"contain"}
			width={"$2.5"}
			{...props}
			source={{ uri: require("../assets/images/catographer-logo.png") }}
		/>
	);
}
