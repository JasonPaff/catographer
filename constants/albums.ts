export const mediaLibraryName = "catographer-app";
