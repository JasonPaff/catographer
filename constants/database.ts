export const databaseName = "catographer-database";
