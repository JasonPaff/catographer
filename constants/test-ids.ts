export const TEST_IDS = {
	catLogoImage: "cat-logo-image",
} as const;
