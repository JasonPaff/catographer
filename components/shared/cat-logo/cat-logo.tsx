import { Image, type ImageStyle, StyleSheet } from "react-native";
import { TEST_IDS } from "../../../constants/test-ids";

export function CatLogo(style?: ImageStyle) {
	const catImage = require("../../../assets/images/catographer-logo.png");
	return (
		<Image
			alt={"catographer logo"}
			source={catImage}
			style={{ ...styles.image, ...style }}
			testID={TEST_IDS.catLogoImage}
		/>
	);
}

const styles = StyleSheet.create({
	image: {
		height: 35,
		objectFit: "contain",
		width: 35,
	},
});
