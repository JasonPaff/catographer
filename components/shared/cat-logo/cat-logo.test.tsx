import { render, screen } from "@testing-library/react-native";
import { TEST_IDS } from "../../../constants/test-ids";
import { CatLogo } from "./cat-logo";

test("<CatLogo/> should render correctly", () => {
	render(<CatLogo />).toJSON();
	expect(screen).toMatchSnapshot();
});

test("<CatLogo/> should have alt attribute", () => {
	render(<CatLogo />);
	const image = screen.getByTestId(TEST_IDS.catLogoImage);
	expect(image).toHaveProperty("_fiber.memoizedProps.alt", "catographer logo");
});
