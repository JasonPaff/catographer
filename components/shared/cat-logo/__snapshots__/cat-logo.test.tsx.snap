// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatLogo/> should render correctly 1`] = `
<Image
  alt="catographer logo"
  source={1}
  style={
    {
      "height": 35,
      "objectFit": "contain",
      "width": 35,
    }
  }
  testID="cat-logo-image"
/>
`;
